### PR TITLE
Dynamically write dataset_description.json file to suppress pybids warning

### PR DESCRIPTION
**Pull request by gisellemartel**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [x] I've assigned a reviewer

## Description
If the derivatives option is set to true in the new loader, this change will check if a `dataset_description.json` file was included in the derivatives subfolder. When it is not included, pybids issues a warning. The purpose of this change is to remove this warning by writing a default `dataset_description.json` file in the derivatives subfolder if it does not exist.

## Linked issues
#660

### CODE REVIEW

File: ivadomed/loader/utils.py [hunk 1]
```diff
@@ -918,7 +921,7 @@ def create_bids_dataframe(self):
                 self.df = self.df[self.df['filename'].str.contains('|'.join(has_deriv))
                                   | self.df['filename'].str.contains('|'.join(deriv))]
             else:
-                # Raise error and exit if no derivatives are found for any subject files
+                 # Raise error and exit if no derivatives are found for any subject files
```
Comment (andreanne-lemay) [inline ivadomed/loader/utils.py:924]:
```suggestion
                # Raise error and exit if no derivatives are found for any subject files
```

**Review by andreanne-lemay — APPROVED** [review]
LGTM :) Thank you for taking care of this!

File: ivadomed/loader/utils.py (hunk 1 above)
Comment (mariehbourget) [inline ivadomed/loader/utils.py:924]:
```suggestion
                # Raise error and exit if no derivatives are found for any subject files
```

**Review by mariehbourget — APPROVED** [review]
Looks good! Thanks @gisellemartel for taking care of this issue.

**Pull request merged**